foundry/applications/formApplications/permissionConfig.d.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * An application for configuring the permissions which are available to each User role.
+ */
+declare class PermissionConfig extends FormApplication<PermissionConfig.Data> {
+  /**
+   * @override
+   * @defaultValue
+   * ```typescript
+   * mergeObject(super.defaultOptions, {
+   *   title: game.i18n.localize("PERMISSION.Title"),
+   *   id: "permissions-config",
+   *   template: "templates/sidebar/apps/permission-config.html",
+   *   width: 660,
+   *   height: "auto",
+   *   scrollY: [".permissions-list"],
+   *   closeOnSubmit: true
+   * })
+   * ```
+   */
+  static get defaultOptions(): FormApplication.Options;
+
+  /**
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): PermissionConfig.Data;
+
+  /**
+   * Prepare the permissions object used to render the configuration template
+   */
+  protected _getPermissions(current: typeof CONFIG['User']['permissions']): PermissionConfig.Permission[];
+
+  /** @override */
+  activateListeners(html: JQuery): void;
+
+  /**
+   * Handle button click to reset default settings
+   * @param event - The initial button click event
+   */
+  protected _onResetDefaults(event: JQuery.ClickEvent): void;
+
+  /** @override */
+  protected _onSubmit(
+    event: Event,
+    options?: FormApplication.OnSubmitOptions
+  ): ReturnType<FormApplication['_onSubmit']>;
+
+  /**
+   * @param event - (unused)
+   * @override
+   */
+  protected _updateObject(event: Event, formData: Record<string, boolean>): Promise<void>;
```

Comment:
can we type `formData` more concretely?

Reply:
It would need to be extendable, if I remember correctly. Because you can add permissions, I believe.

Reply:
ok, maybe just create a named type for it anyways? just to be consistent with other FormApplications.